### DISCUSSION TITLE
Dans Mattermost, j'aimerai savoir à qui je parle rapidement (startup, incubateur, email, photo, et bien plus encore)

### DISCUSSION THREAD

**Discussion by gauthierna** [betagouv/cap#13] [Besoin]
### Besoin

```bash
J'échange souvent avec des gens sur Mattermost et je trouve dommage qu'on ne puisse pas savoir rapidement avec qui on parle, ou qui participe à une conversation.
```

### Solutions envisageables

```bash
Pouvoir avoir plus d'infos sur le profil de la personne directement depuis Mattermost.

Pour ça, il faudrait pouvoir "augmenter" le profil quand on clique sur le nom ou la photo, avec les informations genre :
- Rôle
- Startup(s)
- Incubateur
- Dans la communauté beta depuis X années
- Skills / Stacks
- Lien vers sa fiche membre ? 

Ces infos pourraient être récupérées automatiquement via la fiche membre par exemple.

Voici un plugin qui pourrait faire le job ->

https://github.com/mattermost/mattermost-plugin-custom-attributes/tree/master

Remarque : c'est un plugin utilisé par le mattermost de la communauté Mattermost, donc que je juge plutôt fiable (community.mattermost.com)
```

### Informations complémentaires

- [X] J'avais déjà entendu ce besoin au sein de la communauté beta.gouv
- [X] Je me porte volontaire pour participer à la mise en place de la future solution choisie

**Comment by LucasCharrier** [comment]
1. Depuis la semaine dernière les nouveaux membres ont forcément leur role et le produit pour lequel ils travaillent renseignés lors de la création du compte mattermost.
2. On ne peut malheureusement pas facilement augmenté le profil. Les plugins qui existent ne couvrent pas vraiment ces besoins. On avait investigué avec @chaibax le plugin https://github.com/mattermost/mattermost-plugin-custom-attributes/tree/master et il ne permet pas de rajouter des champs personnalisé par utilisateur. Il permet en revanche de mettre en avant le fait qu'un utlisateur appartienne à un groupe. Donc on pourrait rajouter un badge [Dinum] sur les membres de la dinum par exemple. Mais l'édition se fait à la main, je ne sais pas si c'est automatisable.

**Reply by gauthierna** [reply]
1. Trop cool, tu as un exemple d'un user que je peux checker pour voir comment ça se présente ? 

2. D'après ce que je lis, tu peux mettre des groupes de users, ou des users direct (sachant que dans ce qu'ils appellent les labels, tu peux mettre des emojis, des urls...etc. 
Sur l'automatisation, tout se passe dans le config.json, donc j'imagine qu'on peut bricoler un truc qui l'update de manière hebdo par exemple.  cf -> https://github.com/mattermost/mattermost-plugin-custom-attributes/tree/master?tab=readme-ov-file#usage

(bon, je dis ça comme c'était trivial, mais je suis incapable de le faire, sinon j'aurai testé ah ah)

**Reply by revolunet** [reply]
ça doit être possible de regénérer le config.json avec les attributs/valeurs qui vont bien mais il faudrait probablement redémarrer mattermost pour que ca le prenne en compte à chaque fois ?

**Reply by gauthierna** [reply]
Je reprise que je suis "non tech" ah ah mais : quand tu vas éditer un groupe à la volée dans l'interface du plugin, il soit bien écrire en live sur le json sans redémarrer toute l'app ?

Sinon j'ai envie de me chauffer et d'essayer de le faire avec chatGPT ah ah. Ou de faire l'API qui va bien - c'est une issue remontée dans le repo du plugin ;) peut être ma première contribution ever à un projet open source ?

**Reply by revolunet** [reply]
En général, un fichier de conf, il faut relancer l'appli pour qu'il soit pris en compte mais ce n'est p-e pas le cas ici, je vais demander [sur le repo](https://github.com/mattermost/mattermost-plugin-custom-attributes/issues/93).

la contribution c'est toujours instructif comme démarche :)

**Reply by revolunet** [reply]
> If you edit config.json directly, you do need to restart the server. However, if you edit the config using the plugin settings UI, then you won't need to restart the server.

**Reply by LucasCharrier** [reply]
Notre config.json est en bdd il me semble, il faut modifier l'entrée directement dans la bdd. Mais j'ai testé la possibilité de faire eventuellement nous même une app à part qui pourrait interagir comme ce welcome bot.

**Reply by gauthierna** [reply]
@LucasCharrier attention les plugins welcome bot et cet outil "custom attribute" ne sont pas les mêmes.

**Reply by gauthierna** [reply]
PS : du coup s'il est déjà installé (ce que tu disais dans une conv Mattermost), on peut accéder à l'interface pour voir la tête que ça a ?

**Reply by LucasCharrier** [reply]
Yes,  les deux sont déjà installés, mais assez chiant à configurer, il n'y a pas d'interface pour le welcome bot, c'est du `json` en bdd. L'interface pour custom attribute est très limité, seul les admins peuvent y accéder et tu peux juster sélectionner les membres et écrire ce que tu veux afficher sur leur profil. On l'a deja mis en place, il y a un label [Admin Tech] sur mon profil et celui de Julien par exemple. C'est édité à la main, tu peux pas définir une règle par l'interface. En gros ça demande pas mal de temps de dev, si on veut mettre ça en place et que ça rende bien. 
Custom attribute c'est vraiment à oublier mais sur le welcome bot on peut essayer quelque chose.

**Comment by revolunet** [comment]
variante possible : un lien à cliquer qui renvoie vers le profil complet sur l'espace-membre

**Reply by revolunet** [reply]
alternative: testez la commande `/fiche [nom du produit ou personne]` dans mattermost.

ca utilise les données qu'on a, qui sont encore imparfaites, mais on y travaille parrallèlement...

on peut imaginer enrichir le résultat avec d'autres infos, là c'est que des infos publiques.

thoughts ?
